### ISSUE TITLE
Teamcity 9 and OpenCover Runner plugin

### ISSUE THREAD

**Issue by odministrator** [mindriven/TeamCity-OpenCoverRunner#6]
Hello,
I'm trying to configure plugin with latest Teamcity (9.0.2):
I uploaded plugin to Teamcity lib dir and restarted Teamcity server. So, I see plugin in plugins list like:
http://clip2net.com/s/3bOkrF9
but I can't find it in Runners list:
http://clip2net.com/s/3bOkF6e
Maybe i have to configure something?
Help please :)

**Comment by mindriven** [comment]
Hello
Thanks for trying out my plugin.
I didn't check the compatibility with TC9, so problem may lie there (TC API might have changed). Another cause might be Java version. Can you please check TC logs, usually there is something usefull there.
Please let me know about the result.

**Comment by odministrator** [comment]
Hi
I looked at logs, but can't find anything strange. You can find logs at https://www.dropbox.com/s/4ekdiw976gqn1nq/teamcity_logs_default.zip?dl=0
Thank you.

**Comment by mindriven** [comment]
Hi
Sorry for the lag. I've also examined the logs, it seems like TC consumed plugin jar without problem. But what stands out to me is that after this entry:

[2015-02-06 14:34:56,718]   INFO - nt.AgentPluginsVersionProvider - Plugins "agent" part signature is updated: 

OpenCoverRunner is not listed as being updated on an agent. This can be ok, if previously this plugin was already installed with the same version. Was it? If not we may have a lead.

**Comment by odministrator** [comment]
Hi,

No, it was a new TeamCity installation, but I installed plugin several
times.

On Thu, Feb 19, 2015 at 10:21 PM, mindriven <notifications@github.com>
wrote:

> Hi
> Sorry for the lag. I've also examined the logs, it seems like TC consumed
> plugin jar without problem. But what stands out to me is that after this
> entry:
>
> [2015-02-06 14:34:56,718] INFO - nt.AgentPluginsVersionProvider - Plugins
> "agent" part signature is updated:
>
> OpenCoverRunner is not listed as being updated on an agent. This can be
> ok, if previously this plugin was already installed with the same version.
> Was it? If not we may have a lead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mindriven/TeamCity-OpenCoverRunner/issues/6#issuecomment-75117443>
> .
>



-- 
Best regards,
Piotr Roshkovski

GSM: +37529 - 55 868 11
e-mail: odministrator@gmail.com

**Comment by mindriven** [comment]
ok, got it. Whenever I will find a moment I will try to run my plugin against TC 9.0.2 and will let you know about the results.

**Comment by josh3ennett** [comment]
Yeah I had the same issue in TeamCity 9.1 while building the project using JDK v8 .  It has to be build with JDK v7.  I built it successfully from IntelliJ IDEA.  Uploaded it through the TeamCity admin screen.

**Comment by mindriven** [comment]
@josh3ennett: thanks for your contribution.
Is it so, that JDK version against which the plugin is built must be the same that TC is running against, or it just have to be v7? Or is TC bound to run with (possibly packaged) v7 for now?

**Comment by josh3ennett** [comment]
Good point.  When I changed my local TeamCity instance to run against JRE 1.8 and the plugin I built using the JDK v8 worked fine.  I also tried deploying the version I built using the JDK v7 to the TeamCity instance running JRE 1.8  and that worked also.

**Comment by dklevakin** [comment]
I have the same issue. 
Found interesting bug:
If I add /tools into path to OpenCover the search fails. Output: Search *
The default one works fine - packages/OpenCover*/OpenCover.Console.exe
But in new NuGet package Opencover is placed to /tools folder.
